fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -561,6 +561,18 @@ class ApplyMetadataMutationsImpl {
 		TEST(true); // Recovering at a higher version.
 	}
 
+	void checkSetVersionEpochKey(MutationRef m) {
+		if (m.param1 != versionEpochKey) {
+			return;
+		}
+		int64_t versionEpoch = BinaryReader::fromStringRef<int64_t>(m.param2, Unversioned());
+		TraceEvent("VersionEpoch", dbgid).detail("Epoch", versionEpoch);
+		if (!initialCommit)
+			txnStateStore->set(KeyValueRef(m.param1, m.param2));
+		confChange = true;
+		TEST(true); // Setting version epoch
```

Comment:
A recovery in the case where the version epoch is made older allows us to advance the version appropriately. Does this also give a benefit if we move it in the other direction?

Reply:
No, but the message the master uses to update its version epoch is only sent from the cluster recovery path (`UpdateRecoveryDataRequest`), and I'm not sure it's worth adding logic to avoid a recovery only when increasing the version epoch (assuming it's a somewhat rare operation).